crates/metaslang/cst/src/cursor.rs

```diff
@@ -6,10 +6,34 @@ use crate::kinds::KindTypes;
 use crate::nodes::{Edge, Node, NonterminalNode};
 use crate::text_index::{TextIndex, TextRange};
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+enum Parent<T: KindTypes> {
```

Comment:
rather than introducing the new `Parent` abstraction, WDYT of adding a `PathAncestor::is_included: bool` field instead, and keeping the `Option<Rc<PathAncestor<T>>>`? since that most callsites of this type don't care about that distinction...